streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -404,15 +404,12 @@ private void shutdown() {
         } catch (Throwable e) {
             log.error("{} Failed to close restore consumer: ", logPrefix, e);
         }
-
-        // TODO remove this
-        // hotfix to improve ZK behavior als long as KAFKA-4060 is not fixed (c.f. KAFKA-4369)
-        // when removing this, make StreamPartitionAssignor#internalTopicManager "private" again
-        if (partitionAssignor != null && partitionAssignor.internalTopicManager != null) {
-            partitionAssignor.internalTopicManager.zkClient.close();
+        try {
+            partitionAssignor.close();
+        } catch (Throwable e) {
+            log.error("stream-thread [{}] Failed to close KafkaStreamClient: ", this.getName(), e);
```

Comment:
I think this is not a good pattern, since AFAIK unlike the other clients the only `throwable` is actually `IOException` (your error message also indicates that doesn't it :P `KafkaStreamClient`), and it is only thrown from the metricsReporter.close(), capturing all throwable may hide some issues. Instead, we can just let `InternalTopicManager.close()` to capture and log if there is any `IOException` and not throw it all the way here.